test/common/quic/platform/quic_platform_test.cc

```diff
@@ -144,7 +144,8 @@ TEST_F(QuicPlatformTest, QuicExportedStats) {
 
 TEST_F(QuicPlatformTest, QuicHostnameUtils) {
   EXPECT_FALSE(QuicHostnameUtils::IsValidSNI("!!"));
-  EXPECT_FALSE(QuicHostnameUtils::IsValidSNI("envoyproxy"));
+  // SNI without dot is valid as per RFC 2396.
```

Comment:
I think RFC 2396 is a URL RFC, but this code is about valid SNIs, right? Should we quote 3546? That being said doesn't 3546 talk about fully qualified (and hence dot-containing)?

In any case, I'm mildly surprised that there is a behavior change here. The .h file for the old impl has:


  // Returns true if the sni is valid, false otherwise.
  //  (1) disallow IP addresses;
  //  (2) check that the hostname contains valid characters only; and
  //  (3) contains at least one dot.

It sure looks like it's explicitly checking for a dot. I don't see anything in the CL description about this behavior change. Do you know if this was accidental or intentional?

Reply:
Looks like this was an intentional QUICHE change, see internal cl/363489827 for details. This PR just makes Envoy match QUICHE now that it's no longer possible to override this API.

Reply:
Ah! Thanks for that context. I failed to find it. Ok, that makes sense. (Should we reference RFC 3546 instead of RFC 2396 in that comment?)